bus-mapping/src/evm/opcodes/add.rs

```diff
@@ -0,0 +1,217 @@
+            StackOp::new(RW::WRITE, stack_second_last_position, sum),
+        );
+
+        exec_step.mut_stack().add();
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod add_tests {
+    use super::*;
+    use crate::{
+        bytecode,
+        evm::{
+            EvmWord, GasCost, GlobalCounter, Memory, OpcodeId,
+            OpcodeId::{ADD, PUSH1},
+            Stack, StackAddress, Storage,
+        },
+        external_tracer,
+        operation::RW::{READ, WRITE},
+        BlockConstants, ExecutionTrace,
+    };
+    use pasta_curves::pallas::Scalar;
+
+    fn step_setup(
+        stack: Stack,
+        instruction: OpcodeId,
+        obtained_steps: ExecutionStep,
+        gc: GlobalCounter,
+    ) -> ExecutionStep {
+        ExecutionStep {
+            memory: Memory::empty(),
+            stack,
+            storage: Storage::empty(),
+            instruction: instruction,
+            gas: obtained_steps.gas(),
+            gas_cost: GasCost::FASTEST,
+            depth: 1u8,
+            pc: obtained_steps.pc(),
+            gc: gc,
+            bus_mapping_instance: vec![],
+        }
+    }
+
+    #[test]
+    fn add_opcode_impl() -> Result<(), Error> {
+        let code = bytecode! {
+            #[start]
+            PUSH1(0x80u64)
+            PUSH1(0x80u64)
+            ADD
+            STOP
+        };
+
+        let block_ctants = BlockConstants::default();
+
+        // Get the execution steps from the external tracer
+        let obtained_steps = &external_tracer::trace(&block_ctants, &code)?
+            [code.get_pos("start")..];
+
+        // Obtained trace computation
+        let obtained_exec_trace = ExecutionTrace::<Scalar>::new(
+            obtained_steps.to_vec(),
+            block_ctants,
+        )?;
+
+        let mut ctx = TraceContext::new();
+        let last_stack_pointer = StackAddress::from(1022);
+        let second_last_stack_pointer = StackAddress::from(1023);
+        let stack_value_a = EvmWord::from(0x80u8);
+        let stack_value_b = EvmWord::from(0x80u8);
+        let sum = stack_value_a.adc(stack_value_b).unwrap();
+
+        // Generate Step1 corresponding to PUSH1 80
+        let mut step_1 = step_setup(
+            Stack::empty(),
+            PUSH1,
+            obtained_steps[0].clone(),
+            ctx.gc,
+        );
+
+        // Add StackOp associated to the 0x80 push at the latest Stack pos.
+        ctx.push_op(
+            &mut step_1,
+            StackOp::new(WRITE, second_last_stack_pointer, stack_value_a),
+        );
+
+        // Generate Step2 corresponding to PUSH1 80
+        let mut step_2 = step_setup(
+            Stack(vec![stack_value_a]),
+            PUSH1,
+            obtained_steps[1].clone(),
+            ctx.gc,
+        );
+
+        // Add StackOp associated to the 0x80 push at the second latest Stack pos.
+        ctx.push_op(
+            &mut step_2,
+            StackOp::new(WRITE, last_stack_pointer, stack_value_b),
+        );
```

Comment:
Following @ed255 This two `ExecutionStep`s are not needed. As they're not included in the trace.

Reply:
Yes I mistook.